lab5/src/main/java/SearchArray.java

```diff
@@ -0,0 +1,149 @@
+/**
+ *	SearchArray implements a linear search in an array, searching
+ *	for a user provided value.
+ *
+ *	Class Invariants:
+ * 	'maxVal' must be greater than 'minVal'
+ *	All elements of the array must lie in the range defined by
+ *	 	'minVal' and 'maxVal' variables
+ *	@author
+ *	@version Mar 3, 2012
+ *	@project CMSC 202 - Spring 2012 - Project #
+ *	@section #
+ *
+ */
+public class SearchArray {
+    private int array[],minVal,maxVal;
+    boolean constructed=false;
+    /**
+     * Assigns the instance variable array reference to
+     * point to the incoming array reference. Also assigns the
+     * range values.
+     * Precondition: The input array to the constructor cannot be null.
+     * @param array The input array to be searched on.
+     * @param minVal The minimum allowed value in the array.
+     * @param maxVal The maximum allowed value in the array.
+     */
+    public SearchArray(int[] array, int minVal, int maxVal) {
+
+        //Precondition checking code
+        if(array==null){
+            throw new NullPointerException("Constructor precondition " +
+                    "not met: Input array cannot be null");
+        }
+
+        //class invariant #1 checking code
+        if(maxVal<minVal){
+            throw new RuntimeException("Class invariant " +
+                    "not met: maxVal cannot be less than minVal");
+        }
+
+        for(int i : array){
+            if (i < minVal || i > maxVal){
+                throw new RuntimeException("Class invariant " +
+                        "not met: All elements of the array must lie in the range defined by" +
+                        " minVal' and 'maxVal' variables");
+            }
+        }
+
+        this.array=array;
+        this.minVal=minVal;
+        this.maxVal=maxVal;
+        constructed=true;
+        System.out.println("\tSearchArray object successfully initialized");
+    }
+
+
+    /**
+     * Method to search for a number in the array.
+     * Precondition: the least possible value of x can be minVal and
+     * 		the max possible value can be maxVal.
+     * @param x The number to be searched
+     */
+    public void search(int x){
+        if(constructed){
+            /*
+             * Enter Pre-condition checking code here
+             */
+            if (x < minVal || x > maxVal){
+                throw new RuntimeException("Search method precondition " +
+                        "not met : the least possible value of x can be minVal and" +
```

Comment:
Either return a "not found" or use actual values. "minVal" wouldn't make a good error message.